src/uart.rs

```diff
@@ -770,6 +777,9 @@ impl<'a> Uart<'a, Async> {
         let tx = tx.into();
         let rx = rx.into();
 
+        T::Interrupt::unpend();
+        unsafe { T::Interrupt::enable() };
+
```

Comment:
Technically this is not needed, as `Uart` builds `UartRx` and `UartTx`, both of which will `unpend()` and `enabled()` the interrupt. Still, if we ever change the driver, maybe this is better.